frontend/src/components/reset-password/reset-password.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Link, useNavigate } from 'react-router-dom';
+
+import { AppRoute } from '@common/enums/app/app-route.enum';
+import { ButtonFill } from '@components/common/button-fill/button-fill';
+import { InputField } from '@components/common/input-field/input-field';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { Alert } from '@mui/material';
+import Container from '@mui/material/Container';
+import { useResetPasswordMutation } from '@store/queries/forgot-password';
+
+import { resetPassword as resetPasswordSchema } from '../../../../shared/src/validation-schemas/validation-schemas';
+import Logo from '../../assets/images/logo.svg';
+import BgImage from '../../assets/images/sign-bg.jpg';
+import styles from './styles.module.scss';
+
+export const ResetPassword = (): React.ReactElement => {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: yupResolver(resetPasswordSchema) });
+  const [resetPassword, { isLoading }] = useResetPasswordMutation();
+  const [backendError, setBackendError] = useState('');
```

Comment:
Mutations already provide the `error` object in the response.
So, instead of using the promises chaining, you can use the `useEffect` hook + RTQK results. 
If `isSuccess=true` -> redirect user to the `'/sign-in'`. Or you can show the error if the `error` object is not `undefined`